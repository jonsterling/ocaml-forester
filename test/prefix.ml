open Forester

(* pretty-print an address prefix, ignoring the index *)
let pp_print_prefix formatter (prefix, _index) =
  Format.pp_print_string formatter prefix

let _ =
  let addrs = List.to_seq [ "foo"; "bar-0001"; "foo-bar-0002"; "2024-05-08"] in
  let prefixes = Forest.prefixes ~addrs in
  Forest.Prefix_map.to_list prefixes
  |> Format.printf "prefixes: @[<hov>%a@]@."
  @@ Format.pp_print_list ~pp_sep:Format.pp_print_space pp_print_prefix
