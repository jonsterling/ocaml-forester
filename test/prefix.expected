prefixes: 2024-05-08 bar foo foo-bar
